### PR TITLE
feat(vector): support VECTOR as a non-key value column in Parquet-backed primary-key tables.

### PR DESCRIPTION
**Pull request by ChaomingZhangCN**
### Purpose

Linked issue: #197 

Support `VECTOR` as a non-key value column in Parquet-backed primary-key tables.

This PR is scoped to PK tables:
- It does not add `VECTOR` support for data-evolution tables.
- It does not add or change schema-evolution APIs; schema evolution remains a separate capability.
- `VECTOR` remains unsupported as a primary, partition, bucket, sequence, or sequence-group ordering field.

The implementation adds FixedSizeList handling across PK writes, spill serialization,
merge/compaction, equality checks, and Arrow read-back.

### Tests

- `git diff --check`
- CMake configuration completed successfully.
- Focused schema-validation and PK VECTOR write/read/compaction tests are included.

> The focused build and test execution is still running locally because Arrow and its
> dependencies must first be built from source.

### API and Format

No public API or on-disk format change.

### Documentation

Updated the VECTOR type documentation to state support for PK-table value columns and
the continued restriction for data-evolution tables.

### Generative AI tooling

Generated-by: Codex (GPT-5)

### CODE REVIEW

File: src/paimon/core/io/row_to_arrow_array_converter.h [hunk 1]
```diff
@@ -158,6 +158,12 @@ Status RowToArrowArrayConverter<T, R>::Reserve(arrow::ArrayBuilder* array_builde
             PAIMON_RETURN_NOT_OK(Reserve(list_builder->value_builder(), idx));
             break;
         }
+        case arrow::Type::type::FIXED_SIZE_LIST: {
+            PAIMON_ASSIGN_OR_RAISE(arrow::FixedSizeListBuilder * list_builder,
```
Comment (lxy-9602) [inline src/paimon/core/io/row_to_arrow_array_converter.h]: May use `auto*` here to avoid formatting issues.

File: src/paimon/core/schema/schema_validation_test.cpp [hunk 2]
```diff
@@ -116,12 +157,6 @@ TEST(SchemaValidationTest, TestVectorType) {
                                              /*primary_keys=*/{}, data_evolution_options));
     ASSERT_NOK_WITH_MSG(SchemaValidation::ValidateTableSchema(*table_schema),
                         "VECTOR fields in data-evolution tables are not implemented yet.");
-    ASSERT_OK_AND_ASSIGN(table_schema,
-                         TableSchema::Create(/*schema_id=*/0, nested_schema,
-                                             /*partition_keys=*/{},
-                                             /*primary_keys=*/{}, data_evolution_options));
-    ASSERT_NOK_WITH_MSG(SchemaValidation::ValidateTableSchema(*table_schema),
-                        "VECTOR fields in data-evolution tables are not implemented yet.");
```
Comment (lxy-9602) [inline src/paimon/core/schema/schema_validation_test.cpp:125]: I’m wondering why this test was removed.

File: test/inte/write_and_read_inte_test.cpp [hunk 3]
```diff
@@ -678,6 +678,60 @@ TEST_P(WriteAndReadInteTest, TestPKSimple) {
     ASSERT_TRUE(success);
 }
 
+TEST_P(WriteAndReadInteTest, TestPKVector) {
+    auto [file_format, file_system] = GetParam();
+    if (file_format != "parquet") {
+        return;
+    }
+
```
Comment (lxy-9602) [inline test/inte/write_and_read_inte_test.cpp:732]: It looks like the current PR is intended to support merge functions for all types. Please update the issue description accordingly, and also add tests for listagg and spill.

Comment (lxy-9602) [reply test/inte/write_and_read_inte_test.cpp:732]: Also, could you add a test for a PK table with a nested type containing a vector field, to verify the recursive row/column conversion for vectors?

File: src/paimon/core/io/row_to_arrow_array_converter.h (hunk 1 above)
Comment (ChaomingZhangCN) [reply src/paimon/core/io/row_to_arrow_array_converter.h]: ok

File: src/paimon/core/io/key_value_in_memory_record_reader.cpp [hunk 4]
```diff
@@ -111,22 +111,42 @@ void KeyValueInMemoryRecordReader::Close() {
...
     }
     auto sort_options =
         arrow::compute::SortOptions(sort_keys, arrow::compute::NullPlacement::AtStart);
     arrow::compute::ExecContext exec_context(arrow_pool_.get());
-    PAIMON_ASSIGN_OR_RAISE_FROM_ARROW(std::shared_ptr<arrow::Array> sorted_indices,
-                                      arrow::compute::SortIndices(arrow::Datum(value_struct_array_),
-                                                                  sort_options, &exec_context));
+    // Arrow's StructArray sorting path may inspect value columns outside the sort keys. Restrict
+    // the batch to the requested fields so non-sortable values, such as VECTOR, are never compared.
+    std::shared_ptr<arrow::RecordBatch> sort_batch =
```
Comment (lxy-9602) [inline src/paimon/core/io/key_value_in_memory_record_reader.cpp]: I’d like to better understand what issues `inspect value columns` can cause here. Does it lead to incorrect sorting, crashes, or explicit errors? Also, is this considered a bug in Arrow, and is there any ongoing fix for it?

File: test/inte/write_and_read_inte_test.cpp [hunk 5]
```diff
@@ -69,6 +70,180 @@
 #include "rapidjson/writer.h"
 
 namespace paimon::test {
+namespace {
+
+struct PrimaryKeyVectorRow {
+    int64_t primary_key;
+    std::optional<std::array<float, 3>> embedding;
```
Comment (lxy-9602) [inline test/inte/write_and_read_inte_test.cpp]: Please try to avoid using this kind of struct unless the entire test uses a fixed schema throughout. Otherwise, please use simple Arrow JSON to initialize the input and expected Arrow arrays, as it is more straightforward and easier to read.

File: src/paimon/core/schema/schema_validation_test.cpp (hunk 2 above)
Comment (ChaomingZhangCN) [reply src/paimon/core/schema/schema_validation_test.cpp:125]: Restored in the current revision.

File: test/inte/write_and_read_inte_test.cpp (hunk 5 above)
Comment (ChaomingZhangCN) [reply test/inte/write_and_read_inte_test.cpp]: ok

File: test/inte/write_and_read_inte_test.cpp (hunk 3 above)
Comment (ChaomingZhangCN) [reply test/inte/write_and_read_inte_test.cpp:732]: ok

File: src/paimon/core/io/key_value_in_memory_record_reader.cpp (hunk 4 above)
Comment (ChaomingZhangCN) [reply src/paimon/core/io/key_value_in_memory_record_reader.cpp]: Thanks for questioning this. After checking the Arrow 17 implementation and validating the behavior, I found that my original comment was inaccurate.

  With explicit sort keys, SortIndices converts the StructArray to a RecordBatch and resolves only the requested sort fields. Non-key VECTOR value columns are not used for
  comparison. I have not reproduced incorrect sorting, crashes, or explicit errors caused by those value columns in this path, so I do not have evidence of an Arrow bug or an
  upstream fix to reference.

  I removed the unnecessary key-only projection from both the in-memory PK reader and the realtime PK writer in 0313134. I also added tests covering duplicate keys, stable
  ordering, ascending/descending sequence fields, null and nested VECTOR values, and realtime PK writes.

  The relevant tests passed locally using statically linked test executables to avoid a separate macOS mixed-linking issue.

**Review by lxy-9602 — DISMISSED** [review]
+1

**Review by lxy-9602 — APPROVED** [review]
+1
